rafx-framework/src/render_features/jobs/prepare/view_phase_submit_node_block.rs

```diff
@@ -0,0 +1,118 @@
+use crate::render_features::render_features_prelude::*;
+
+/// The `ID` of a `SubmitNode` in a specific `RenderFeature` in the current frame.
+pub type SubmitNodeId = u32;
+
+/// A generic key usable by the `SubmitNodeSortFunction` for the purpose of sorting the collection of
+/// `RenderFeatureSubmitNode` in a particular `ViewPhase`. This can be used to minimize state changes
+/// in the rendering pipeline, e.g. by setting the bits of the key so that higher bits represent more
+/// expensive state changes like shaders and lower bits represent cheaper state changes like uniforms.
+///
+/// Example: https://web.archive.org/web/20210110113523/https://realtimecollisiondetection.net/blog/?p=86
+pub type SubmitNodeSortKey = u32;
+
+/// The sort function used by a particular `RenderPhase`. This is usually one of the following:
+/// 1. front-to-back
+/// 2. back-to-front
+/// 3. by feature index
+/// 4. unsorted
+pub type SubmitNodeSortFunction = fn(&mut Vec<RenderFeatureSubmitNode>);
```

Comment:
Maybe one day this would be an enum with a few pre-implemented options + a callback for doing something custom